src/plugins/explore/public/components/patterns_table/constants.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HorizontalAlignment } from '@elastic/charts';
+import { PatternItem } from './patterns_table';
+
+export const PATTERNS_FIELD = 'patterns_field';
+export const COUNT_FIELD = 'count';
+
+export const patternsTableColumns = [
+  {
+    field: 'ratio',
+    name: 'Event ratio',
+    render: (val: number) => {
+      // Check if value is a valid, finite number before formatting
+      if (typeof val !== 'number' || isNaN(val) || !isFinite(val)) {
+        return '-';
+      }
+      return `${(val * 100).toFixed(2)}%`;
+    },
+    width: '15%',
+  },
+  {
+    field: 'pattern',
+    name: 'Pattern',
+    render: (pattern: string) => pattern || '-',
+  },
+  {
+    field: 'count',
+    name: 'Event count',
+    render: (val: number) => {
+      // Check if value is a valid, finite number
+      if (typeof val !== 'number' || isNaN(val) || !isFinite(val)) {
+        return '-';
+      }
+      return val;
+    },
+    align: HorizontalAlignment.Right,
+  },
+];
+
+// Mock data for patterns
+export const mockPatternItems: PatternItem[] = [
```

Comment:
can these be defined in storybook rather than mixed with code that will run?

Reply:
you could have like `mocks.ts` or something within this folder that could be used by both the unit tests and storybook. that way its clear that it is purely for dev purposes, `constants.tsx` sounds like it should be used in prod.

Also why is this file `.tsx`? i think `.ts` is sufficient?